lib/cisco_node_utils/bridge_domain_vni.rb

```diff
@@ -0,0 +1,165 @@
+# NXAPI implementation of BridgeDomainVNI class
+#
+# February 2016, Rohan Gandhi Korlepara
+#
+# Copyright (c) 2015-2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+require_relative 'feature'
+
+module Cisco
+  # node_utils class for bridge_domain_vni
+  class BridgeDomainVNI < NodeUtil
+    attr_reader :bd_ids, :bd_ids_list
+
+    def initialize(bds, instantiate=true)
+      # Spaces are removed as bridge-domain cli doesn't accept value with
+      # space
+      @bd_ids = bds.to_s.gsub(/\s+/, '')
+      fail 'bridge-domain value is empty' if @bd_ids.empty? # no empty strings
+
+      @bd_ids_list = Utils.string_to_array(@bd_ids)
+      create if instantiate
+    end
+
+    def to_s
+      "Bridge Domain #{bd_ids}"
+    end
+
+    def self.rangebds
+      hash = {}
+      bd_list = config_get('bridge_domain_vni', 'range_bds')
+      return hash if bd_list.nil?
+
+      bd_list.each do |id|
+        hash[id] = BridgeDomainVNI.new(id, false)
+      end
+      hash
+    end
+
+    def curr_bd_vni_hash
+      final_bd_vni = {}
+      curr_vni = config_get('bridge_domain_vni', 'member_vni')
+      curr_bd_vni = config_get('bridge_domain_vni', 'member_vni_bd')
+      return final_bd_vni if curr_vni.empty? || curr_bd_vni.empty?
+
+      curr_vni_list = Utils.string_to_array(curr_vni)
+      curr_bd_vni_list = Utils.string_to_array(curr_bd_vni)
+
+      hash_map = Hash[curr_bd_vni_list.zip(curr_vni_list.map)]
+      @bd_ids_list.each do |bd|
+        final_bd_vni[bd.to_i] = hash_map[bd.to_i] if hash_map.key?(bd.to_i)
+      end
+      final_bd_vni
+    end
+
+    # This function will first add bds to the system bridge-domain and then
+    # create the bds. If bds already existing then just create. Else add the
+    # non added bds to system range first then create all. This is to avoid the
+    # idempotency issue as system add command throws error if a bd is already
+    # present in the system range.
+    def create
+      sys_bds_array = Utils.string_to_array(system_bd_add)
+      if (@bd_ids_list - sys_bds_array).any?
+        add_bds = Utils
+                  .unsorted_list_to_string(@bd_ids_list - sys_bds_array)
+        config_set('bridge_domain_vni', 'system_bd_add', addbd: add_bds)
+      end
+      config_set('bridge_domain_vni', 'create', crbd: @bd_ids)
```

Comment:
No during create of bds there is no need for feature vni or vni cli. Its required only during member_vni for which there is a feature enable.
And bridge-domain is not dependent on vni, it can be created any time by the user.